src/migrateCalculatedMeasures.ts

```diff
@@ -1,41 +1,89 @@
+import { getCalculatedMeasures } from "./getCalculatedMeasures";
+import { generateId } from "./generateId";
+
+export interface LegacyCalculatedMeasure {
+  expression: string;
+  formatStringExpression: string;
+  uniqueName: string;
+  owners: string[];
+  readers: string[];
+  isDirectory: boolean;
+}
 
 /**
  * Returns the converted content record tree for the calculted measures, ready to be used in ActiveUI 5.
  */
-export function migrateCalculatedMeasures(
-  legacyCalculatedMeasures: LegacyCalculatedMeasure[],
-  calculatedMeasuresFolder: ContentRecord
-): ContentRecord {
-  const migratedCalculatedMeasures = legacyCalculatedMeasures.map(
-    (legacyMeasure) => migrateCalculatedMeasure(legacyMeasure)
-  );
+export async function migrateCalculatedMeasures(
+  legacyPivotFolder: ContentRecord
+): Promise<ContentRecord> {
+  const entitlements = legacyPivotFolder?.children?.entitlements;
+  const calculatedMeasuresFolder = entitlements?.children?.cm!;
 
-  return {
-    entry: {
-      ...calculatedMeasuresFolder.entry,
-    },
-    children: {
-      content: {
+  const legacyCalculatedMeasures = calculatedMeasuresFolder
+    ? await getCalculatedMeasures(calculatedMeasuresFolder)
+    : [];
+
+  const contentFolder = {
+    entry: { isDirectory: true, owners: ["ROLE_USER"], readers: ["ROLE_USER"] },
+    children: {},
+  };
+
+  const structureFolder = {
+    entry: { isDirectory: true, owners: ["ROLE_USER"], readers: ["ROLE_USER"] },
+    children: {},
+  };
+
+  legacyCalculatedMeasures.forEach((legacyMeasure) => {
+    const { owners, readers, expression, formatStringExpression, uniqueName } =
+      legacyMeasure;
+
+    const id = generateId();
+
+    contentFolder.children = {
+      ...contentFolder.children,
+      [id]: {
         entry: {
-          ...calculatedMeasuresFolder.entry,
+          owners,
+          readers,
+          isDirectory: false,
+          content: JSON.stringify({
+            expression,
+            properties: [`FORMAT_STRING=${formatStringExpression}`],
+          }),
         },
-        children: migratedCalculatedMeasures.reduce((acc, currentCM) => {
-          return { ...acc, ...currentCM.content };
-        }, {}),
       },
-      structure: {
+    };
+
+    structureFolder.children = {
+      ...structureFolder.children,
+      [id]: {
         entry: {
-          ...calculatedMeasuresFolder.entry,
+          isDirectory: true,
+          owners,
+          readers,
+        },
+        children: {
+          [`${id}_metadata`]: {
+            entry: {
+              content: JSON.stringify({ name: uniqueName }),
+              owners,
+              readers,
+              isDirectory: false,
+            },
+          },
         },
-        children: migratedCalculatedMeasures.reduce((acc, currentCM) => {
-          return { ...acc, ...currentCM.structure };
-        }, {}),
       },
+    };
+  });
+
+  return {
+    entry: {
+      ...calculatedMeasuresFolder.entry,
+    },
```

Comment:
Here as I mentioned before I don't think we should rely on the permissions of `/pivot/entitlements/cm`: with the ActiveUI 5 structure, we should just define a new directory entry with `ROLE_CS_ROOT` both in `readers` and `owners.`